packages/flutter/lib/src/cupertino/route.dart

```diff
@@ -389,97 +395,96 @@ class CupertinoPageTransition extends StatefulWidget {
     required this.linearTransition,
   });
 
-  /// The widget below this widget in the tree.
-  final Widget child;
-
-  ///  * `primaryRouteAnimation` is a linear route animation from 0.0 to 1.0
-  ///    when this screen is being pushed.
+  /// A linear route animation from 0.0 to 1.0 when this screen is being pushed.
   final Animation<double> primaryRouteAnimation;
 
-  ///  * `secondaryRouteAnimation` is a linear route animation from 0.0 to 1.0
-  ///    when another screen is being pushed on top of this one.
+  /// A linear route animation from 0.0 to 1.0 when another screen is being
+  /// pushed on top of this one.
   final Animation<double> secondaryRouteAnimation;
 
-  ///  * `linearTransition` is whether to perform the transitions linearly.
-  ///    Used to precisely track back gesture drags.
+  /// Whether to perform the transitions linearly. Used to precisely track back
+  /// gesture drags.
   final bool linearTransition;
 
+
+  /// The widget below this widget in the tree.
+  final Widget child;
+
   @override
   State<CupertinoPageTransition> createState() => _CupertinoPageTransitionState();
 }
 
 class _CupertinoPageTransitionState extends State<CupertinoPageTransition> {
 
+  CurvedAnimation? _primaryPositionCurvedAnimation;
 
-  // When this page is coming in to cover another page.
+  /// When this page is coming in to cover another page.
   late Animation<Offset> _primaryPositionAnimation;
-  // When this page is becoming covered by another page.
+
+  CurvedAnimation? _secondaryPositionCurvedAnimation;
+
+  /// When this page is becoming covered by another page.
   late Animation<Offset> _secondaryPositionAnimation;
-  // Shadow of page which is coming in to cover another page.
+
+  CurvedAnimation? _primaryShadowCurvedAnimation;
   late Animation<Decoration> _primaryShadowAnimation;
-  // Curve of primary page which is coming in to cover another page.
-  CurvedAnimation? _primaryPositionCurve;
-  // Curve of secondary page which is becoming covered by another page.
-  CurvedAnimation? _secondaryPositionCurve;
-  // Curve of primary page's shadow.
-  CurvedAnimation? _primaryShadowCurve;
 
   @override
-  void initState() {
+  void initState()  {
     super.initState();
-    _setupAnimation();
+    _setAnimations();
   }
 
   @override
-  void didUpdateWidget(covariant CupertinoPageTransition oldWidget) {
+  void didUpdateWidget(CupertinoPageTransition oldWidget) {
     super.didUpdateWidget(oldWidget);
-    if (oldWidget.primaryRouteAnimation != widget.primaryRouteAnimation
-    || oldWidget.secondaryRouteAnimation != widget.secondaryRouteAnimation
-    || oldWidget.child != widget.child || oldWidget.linearTransition != widget.linearTransition) {
-    _disposeCurve();
-    _setupAnimation();
+    if (
```

Comment:
If child changes, why would be not update anymore?

Reply:
`_setUpAnination()` doesn't use `widget.child` so there is no reason to recompute it when `child` changes.

Moreover, `SizedBox() != SizedBox()` is `true`, so that means that `oldWidget.child != widget.child` will most likely be always true